File: app/src/main/java/com/java/eventhike/model/Location.java
package com.java.eventhike.model;

import com.google.gson.annotations.SerializedName;

public class Location {

	@SerializedName("venueName")
	private String venueName;

	@SerializedName("venueId")
	private String venueId;

	@SerializedName("latitude")
	private double latitude;

	@SerializedName("name")
	private String name;

	@SerializedName("venueDetail")
	private String venueDetail;

	@SerializedName("venueImageUrl")
	private String venueImageUrl;

	@SerializedName("longitude")
	private double longitude;

	public void setVenueName(String venueName){
		this.venueName = venueName;
	}

	public String getVenueName(){
		return venueName;
	}

	public void setVenueId(String venueId){
		this.venueId = venueId;
	}

	public String getVenueId(){
		return venueId;
	}

	public void setLatitude(double latitude){
		this.latitude = latitude;
	}

	public double getLatitude(){
		return latitude;
	}

	public void setName(String name){
		this.name = name;
	}

	public String getName(){
		return name;
	}

	public void setVenueDetail(String venueDetail){
		this.venueDetail = venueDetail;
	}

	public String getVenueDetail(){
		return venueDetail;
	}

	public void setVenueImageUrl(String venueImageUrl){
		this.venueImageUrl = venueImageUrl;
	}

	public String getVenueImageUrl(){
		return venueImageUrl;
	}

	public void setLongitude(double longitude){
		this.longitude = longitude;
	}

	public double getLongitude(){
		return longitude;
	}
}

File: app/src/main/java/com/java/eventhike/model/Location.kt
package com.java.eventhike.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

data class Location(

	@field:SerializedName("zip")
	val zip: String? = null,

	@field:SerializedName("country")
	val country: String? = null,

	@field:SerializedName("city")
	val city: String? = null,

	@field:SerializedName("street")
	val street: String? = null,

	@field:SerializedName("latitude")
	val latitude: Double? = null,

	@field:SerializedName("state")
	val state: String? = null,

	@field:SerializedName("longitude")
	val longitude: Double? = null
): Parcelable {
	constructor(parcel: Parcel) : this(
			parcel.readString(),
			parcel.readString(),
			parcel.readString(),
			parcel.readString(),
			parcel.readValue(Double::class.java.classLoader) as? Double,
			parcel.readString(),
			parcel.readValue(Double::class.java.classLoader) as? Double) {
	}

	override fun writeToParcel(parcel: Parcel, flags: Int) {
		parcel.writeString(zip)
		parcel.writeString(country)
		parcel.writeString(city)
		parcel.writeString(street)
		parcel.writeValue(latitude)
		parcel.writeString(state)
		parcel.writeValue(longitude)
	}

	override fun describeContents(): Int {
		return 0
	}

	companion object CREATOR : Parcelable.Creator<Location> {
		override fun createFromParcel(parcel: Parcel): Location {
			return Location(parcel)
		}

		override fun newArray(size: Int): Array<Location?> {
			return arrayOfNulls(size)
		}
	}
}